tests/jest/components/AdminSettings.spec.js

```diff
@@ -727,6 +793,674 @@ describe('AdminSettings.vue', () => {
 		})
 	})
 
+	describe('Managed project folder form (Group Folder Setup)', () => {
+		describe('view mode', () => {
+			describe('without group folder setup', () => {
+				it('should show status as "Inactive"', () => {
+					const wrapper = getWrapper({
+						state: {
+							openproject_instance_url: 'http://openproject.com',
+							openproject_client_id: 'some-client-id-here',
+							openproject_client_secret: 'some-client-secret-here',
+							nc_oauth_client: {
+								nextcloud_client_id: 'some-nc-client-id-here',
+								nextcloud_client_secret: 'some-nc-client-secret-here',
+							},
+							fresh_group_folder_setup: true,
+							// group folder is already not set up
+							group_folder_status: {
+								status: false,
+							},
+							app_password_set: false,
+						},
+					})
+					const groupFolderStatus = wrapper.find(selectors.groupFolderStatus)
+					const actualFolderStatusValue = groupFolderStatus.text()
+					expect(actualFolderStatusValue).toContain('Inactive')
+					expect(wrapper.find(selectors.groupFolderSetupForm)).toMatchSnapshot()
+				})
+			})
+		})
+
+		describe('edit mode ', () => {
+			describe('fresh setup/after reset', () => {
+				beforeEach(async () => {
+					axios.put.mockReset()
+					axios.get.mockReset()
+				})
+
+				it('should show the switch as "On" and button text label as "Setup OpenProject user, group and folder"', async () => {
+					const wrapper = getWrapper({
+						state: {
+							openproject_instance_url: 'http://openproject.com',
+							openproject_client_id: 'some-client-id-here',
+							openproject_client_secret: 'some-client-secret-here',
+							nc_oauth_client: {
+								nextcloud_client_id: 'some-nc-client-id-here',
+								nextcloud_client_secret: 'some-nc-client-secret-here',
+							},
+							fresh_group_folder_setup: true,
+							// group folder is already not set up
+							group_folder_status: {
+								status: false,
+							},
+							app_password_set: false,
+						},
+					})
+					await wrapper.setData({
+						formMode: {
+							groupFolderSetUp: F_MODES.EDIT,
+						},
+					})
+					expect(wrapper.vm.isProjectFolderSwitchEnabled).toBe(true)
+					const setupGroupFolderButton = wrapper.find(selectors.completeGroupFolderSetupWithGroupFolderButton)
+					expect(setupGroupFolderButton.text()).toBe('Setup OpenProject user, group and folder')
+				})
+
+				it('on trigger switch should show button text label "Complete without group folder setup"', async () => {
```

Comment:
is this appropriate for this case?
```suggestion
				it(' should show button text label "Complete without group folder setup" when the switch is "off"', async () => {
```